File: ios/BankShared/Sources/BankShared/Models/Account.swift
import Foundation

public struct Account: Codable, Identifiable, Sendable {
    public let id: Int64
    public let userId: Int64
    public let currency: Currency
    public let balance: Decimal
    public let accountType: AccountType
    public let isClosed: Bool
    public let createdAt: String
}

File: frontend/src/entities/account.ts
import type { AccountType, Currency } from './common'

export interface AccountResponse {
  id: number
  userId: number
  currency: Currency
  balance: number
  accountType: AccountType
  isClosed: boolean
  createdAt: string
}

export interface CreateAccountRequest {
  userId: number
  currency: Currency
}

export interface MoneyOperationRequest {
  amount: number
}
